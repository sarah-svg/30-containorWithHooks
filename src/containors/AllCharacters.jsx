import React, { useState, useEffect } from 'react';
import CharacterList from '../components/characters/CharacterList.jsx';
import { getCharacters } from '../services/fetchApi.js';

const AllCharacters = () => {
  const [loading, setLoading] = useState(true);
  const [characters, setCharacters] = useState([]);

  useEffect(() => {
    getCharacters().then((characters) => {
      setCharacters(characters);
      setLoading(false);  
      console.log(characters);
    });
  }, []);
  if(loading) return <h1 data-testid="loading">Loading Characters...</h1>;
  // eslint-disable-next-line max-len
  return <div data-testid="main"> <CharacterList characters={characters} /> </div>;
};
export default AllCharacters;


