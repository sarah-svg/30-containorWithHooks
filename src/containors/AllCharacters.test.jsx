
global.fetch = require('node-fetch');

import React from 'react';
import { render, screen, cleanup, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';

import AllCharacters from './AllCharacters';

describe('Main component', () => {
  afterEach(() => cleanup());
  it('tests that loading behavior works as expected', () => {
    render(<AllCharacters />);

    const loading = screen.getByTestId('loading');

    return waitFor(() => {
      expect(loading).toHaveTextContent('Loading Characters...');
    });
  });
  // it.only('test the display', async() => {
  //   render(<AllCharacters/>);

  //   const main = screen.getByTestId('main');
  //   const list = screen.getByTestId('list');
 
  //   await setTimeout(() => {
  //     return waitFor(() => {
  //       expect(main).toContainElement(list);
  //     });
  //   }, 5000);


  // });


});
