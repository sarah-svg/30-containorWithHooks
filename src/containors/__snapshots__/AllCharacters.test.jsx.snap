// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component creates a snapshot test of the character containor 1`] = `
<DocumentFragment>
  <div
    text="Loading Characters..."
  >
    Loading Characters...
  </div>
</DocumentFragment>
`;
